### PR TITLE
Feature github_repository_file data source

### PR DESCRIPTION
**Pull request by chriskuchin**

### CODE REVIEW

File: github/resource_github_repository_file.go [hunk 1]
```diff
@@ -3,7 +3,6 @@ package github
 import (
 	"context"
 	"log"
-	"net/http"
```
Comment (du-neuro) [inline github/resource_github_repository_file.go:6]: What's with the deletes here?  Did you find a better way to achieve the same thing?

Comment (chriskuchin) [reply github/resource_github_repository_file.go:6]: Moved it to a different file because i needed those methods in my data source

Comment (chriskuchin) [reply github/resource_github_repository_file.go:6]: You think i should do it differently?

Comment (du-neuro) [reply github/resource_github_repository_file.go:6]: Sorry didn't notice it was just a move.  Thought it had gone away and was confused as to how you were doing the same.

**Review by du-neuro — APPROVED** [review]

**Pull request merged**